src/stdx.zig

```diff
@@ -540,26 +540,133 @@ pub fn hasUniqueRepresentation(comptime T: type) bool {
 
         .Pointer => |info| return info.size != .Slice,
 
-        .Array => |info| return comptime hasUniqueRepresentation(info.child),
+        .Array => |info| return comptime has_unique_representation(info.child),
 
         .Struct => |info| {
-            // Packed structs are always unique.
-            if (info.backing_integer != null) {
-                return true;
+            // Only consider packed structs unique if they are byte aligned.
+            if (info.backing_integer) |backing_integer| {
+                return @sizeOf(@Type(.{ .Struct = info })) * 8 == @bitSizeOf(backing_integer);
```

Comment:
`@sizeOf(T)` would be simpler I think?

Reply:
... yes, I haven't had my Monster yet. :rofl:

Reply:
I think it's time to go pull an espresso too... ☕